client/lib/cart/store/cart-analytics.js

```diff
@@ -1,29 +1,36 @@
 /**
- * Internal dependencies
+ * External dependencies
  */
-var analytics = require( 'lib/analytics' ),
-	cartItems = require( 'lib/cart-values' ).cartItems,
-	difference = require( 'lodash/difference' );
+import { difference, each, isObjectLike, omitBy } from 'lodash';
 
+/**
+ * Internal dependencies
+ */
+import analytics from 'lib/analytics';
+import { cartItems } from 'lib/cart-values';
 import { recordAddToCart } from 'lib/analytics/ad-tracking';
 
 function recordEvents( previousCart, nextCart ) {
-	var previousItems = cartItems.getAll( previousCart ),
+	const previousItems = cartItems.getAll( previousCart ),
 		nextItems = cartItems.getAll( nextCart );
 
-	difference( nextItems, previousItems ).forEach( recordAddEvent );
-	difference( previousItems, nextItems ).forEach( recordRemoveEvent );
+	each( difference( nextItems, previousItems ), recordAddEvent );
+	each( difference( previousItems, nextItems ), recordRemoveEvent );
+}
+
+function flattenCartObject( cartItem ) {
+	return omitBy( cartItem, isObjectLike );
```

Comment:
What do you think about just omitting `extra`, the only object property? This would cause the error to appear again if anyone adds another object property to `cartItem`, but that might be a benefit as it would force them to consider the implications of their change re: analytics.